Formula/gap.rb

```diff
@@ -11,46 +18,127 @@ class Gap < Formula
+      "sgpviz", "singular", "sl2reps", "sla", "smallgrp", "smallsemi",
+      "sonata", "sophus", "spinsym", "standardff", "symbcompcc",
+      "thelma", "tomlib", "toric", "transgrp", "ugaly", "unipot",
+      "unitlib", "utils", "uuid", "walrus", "wedderga", "xmod",
+      "xmodalg", "yangbaxter",
+      # These packages have `doc` and `test` make targets, but we
+      # don't actually call them
+      "4ti2interface", "autodoc", "cap", "examplesforhomalg",
+      "gaussforhomalg", "generalizedmorphismsforcap", "gradedmodules",
+      "gradedringforhomalg", "homalg", "homalgtocas", "io_forhomalg",
+      "linearalgebraforcap", "localizeringforhomalg",
+      "matricesforhomalg", "modulepresentationsforcap", "modules",
+      "monoidalcategories", "nconvex", "ringsforhomalg", "sco",
+      "toolsforhomalg", "toricvarieties",
+    ]
+
+    # These two packages either don't build or don't work
+    # "cddinterface", "xgap",
+    configure_packages = [
+      "anupq", "caratinterface", "crypting", "curlinterface", "cvec",
+      "datastructures", "deepthought", "digraphs", "ferret", "float",
+      "gauss", "grape", "io", "json", "normalizinterface", "nq",
+      "orb", "profiling", "semigroups", "simpcomp", "zeromqinterface",
+      ]
+
+    old_configure_packages = [
+      "ace", "browse", "cohomolo", "edim", "example", "fplsa",
+      "guava", "kbmag",
+    ]
+
+
+
+    all_packages = no_compilation_packages
+                     .concat(configure_packages)
+                     .concat(configure_packages)
+
+    # Start actually building GAP
+    if build.head?
+      system "./autogen.sh"
+    end
+
+    libexec.install Dir["pkg"]
+
+    system "./configure", "--prefix", libexec/"gap", "--with-readline=#{Formula["readline"].opt_prefix}"
+    if build.head?
+      system "make", "bootstrap-pkg-full"
+      # system "make", "doc"      # Do we need this?
     end
+    system "make", "install"
 
     # Create a symlink `bin/gap` from the `gap` binary
-    bin.install_symlink libexec/"gap" => "gap"
+    bin.install_symlink libexec/"gap/bin/gap" => "gap"
 
     ohai "Building included packages. Please be patient, it may take a while"
+    pkg_dir = "#{libexec}/pkg"
+
     cd libexec/"pkg" do
-      # NOTE: This script will build most of the packages that require
-      # compilation. It is known to produce a number of warnings and
-      # error messages, possibly failing to build several packages.
-      system "../bin/BuildPackages.sh", "--with-gaproot=#{libexec}"
+
+      system "mkdir", "#{libexec}/gap/lib/gap/pkg/"
+
+      # The makefiles appear to only be used for docs...
+      # The BuildPackages.sh script didn't call them
+      all_packages.each do |pkg|
+        system "cp", "-R", pkg, "#{libexec}/gap/lib/gap/pkg/"
+        cd pkg do
+          if File.exist?("./prerequisites.sh")
+            system "./prerequisites.sh", "#{libexec}/gap/lib/gap"
+          end
+        end
```

Comment:
Instead of having the `configure_packages` and `configure_packages` lists, here you could have a similar check to see if there is a `./configure` file. If there is, you can distinguish "old" and "new" style by checking of configure contains the string `with-gaproot`...

Though actually, with GAP 4.13.0, *all but one* configure script support `--with-gaproot` -- the one exception is that of `simpcomp`, which however happily ignores it. So you could just ditch the whole `old_configure_packages` part.